src/components/Email/EmailForm.tsx

```diff
@@ -0,0 +1,172 @@
+import { useRouter } from 'next/router';
+import { useCallback, useMemo, useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import { AdditionalContext, useReportError } from '~/contexts/errorReporting/ErrorReportingContext';
+import { useToastActions } from '~/contexts/toast/ToastContext';
+import { EmailFormFragment$key } from '~/generated/EmailFormFragment.graphql';
+import { EmailFormMutation } from '~/generated/EmailFormMutation.graphql';
+import { usePromisifiedMutation } from '~/hooks/usePromisifiedMutation';
+
+import { Button } from '../core/Button/Button';
+import colors from '../core/colors';
+import { HStack, VStack } from '../core/Spacer/Stack';
+
+type Props = {
+  setIsEditMode: (editMode: boolean) => void;
+  queryRef: EmailFormFragment$key;
+};
+
+function EmailForm({ setIsEditMode, queryRef }: Props) {
+  const query = useFragment(
+    graphql`
+      fragment EmailFormFragment on Query {
+        viewer {
+          ... on Viewer {
+            user {
+              id
+            }
+            email {
+              email
+            }
+          }
+        }
+      }
+    `,
+    queryRef
+  );
+
+  const [updateEmail] = usePromisifiedMutation<EmailFormMutation>(graphql`
+    mutation EmailFormMutation($input: UpdateEmailInput!) @raw_response_type {
```

Comment:
nit: don't need the `@raw_response_type` since you're not doing an optimistic update